Hogajama/hogajama-domain/src/main/java/com/gepardec/hogarama/domain/watering/WateringService.java

```diff
@@ -0,0 +1,82 @@
+package com.gepardec.hogarama.domain.watering;
+
+import java.time.LocalDateTime;
+
+import javax.inject.Inject;
+
+import com.gepardec.hogarama.domain.sensor.SensorDAO;
+
+public class WateringService {
+
+	public enum Config{
+
+		DEFAULT(1, 0.2, 5);
+		
+		protected int measureInterval;
+		protected double lowWater;
+		protected int waterDuration;
+		
+		Config( int measureInterval, double lowWater, int waterDuration) {
+			this.waterDuration = waterDuration;
+			this.measureInterval = measureInterval;
+			this.lowWater = lowWater;
+		}
+	}
+
+	@Inject
+	public SensorDAO sensorDao;
+
+	@Inject
+	public WateringConfigDAO configDao;
+
+	@Inject
+	public ActorService actor;
+
+	@Inject
+	public WateringStrategy watering;
+
+	private LocalDateTime now;
+
+	public WateringService() {
+	}
+
+	protected WateringService(SensorDAO sensorDao, ActorService actor, WateringStrategy watering, WateringConfigDAO configDao) {
+		this.sensorDao = sensorDao;
+		this.actor = actor;
+		this.watering = watering;
+		this.configDao = configDao;
+	}
+
+	public void waterAll() {
+		for (String sensorName : sensorDao.getAllSensors()) {
+			int dur = watering.water(getConfig(sensorName), getDate());
+			if (dur > 0) {
+				actor.sendActorMessage(sensorDao.getLocationBySensorName(sensorName), sensorName, dur);
+			}
+		}
+	}
+
+	private WateringConfigData getConfig(String sensorName) {
+		WateringConfigData wconfig = configDao.getBySensorName(sensorName);
+		if ( null != wconfig ) {
+			return wconfig;
+		}
+		
+		wconfig = new WateringConfigData(sensorName, sensorName, Config.DEFAULT.measureInterval,
+				Config.DEFAULT.lowWater, Config.DEFAULT.waterDuration);
```

Comment:
sensorName wird immer gleich dem actorName gesetzt, wird in WateringConfigData dann überhaupt eine eigener Name für den Actor benötigt, da Sensor und Actor ja schlussendlich doch zusammengehören werden?

Reply:
Wir haben jetzt eine 1 - 1 Beziehung Actor - Sensor. Das muss aber nicht so bleiben und ich glaube, man soll Aktoren und Sensoren konzeptionell trennen.